### PR TITLE
Add announcement ranges to the neutral section model

### LINKED ISSUE #4661: BP/CC : Update data model to take into account announcement panels
## Pitch

Either:
* Make the track ranges start at the annoucement, and then give the offset at which the EXE panel is
* Add a track range attribute between annoucement and EXE

## Acceptance criteria

- [ ] Data model is updated everywhere

### PR DESCRIPTION
**Pull request by Castavo**
closes #4661

This PR:
* Adds a new list of track ranges to `NeutralSections`
* Uses it in order to be able to coast on some ranges but still display them in GEV as normaly electrified, as it was requested by PM team. (The electrification in GEV should be refactored some day and some of this would disappear)
* Fixes some tiny misc stuff in front linked to electrification in GEV

This PR may seem a bit large, but most lines of changes comes from the update of `editoast/test/small_infra/small_infra.json`

Please especially test that STDCM works fine, I'm not an expert on this part of the app, and I have some trouble diagnosing if my changes impacted it

### CODE REVIEW

File: core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_utils/RangeMapUtils.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import com.google.common.collect.RangeMap;
 import com.google.common.collect.TreeRangeMap;
 import java.util.function.BiFunction;
+import java.util.function.Function;
```
Comment (clarani) [inline core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_utils/RangeMapUtils.java]:
```suggestion
```
You added only an import here, is it on purpose or is it a mistake ? :)

**Review by clarani — COMMENTED** [review]
Code looks good to me, but I can't test as the backend is broken (500 when I want to create a train, see #4871)

**Review by eckter — APPROVED** [review]

File: core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_utils/RangeMapUtils.java (hunk 1 above)
Comment (Castavo) [reply core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_utils/RangeMapUtils.java]: It's a mistake, thanks

**Review by nicolaswurtz — APPROVED** [review]

**Review by flomonster — APPROVED** [review]
LGTM.

**Pull request merged**